backends/s3/s3.go

```diff
@@ -117,9 +117,18 @@ type Backend struct {
 	lastTime   time.Time
 }
 
-func (b *Backend) List(ctx context.Context, prefix string) (simpleblob.BlobList, error) {
-	// Prepend global prefix
+func (b *Backend) List(ctx context.Context, prefix string) (blobList simpleblob.BlobList, err error) {
+	// Handle global prefix
 	prefix = b.prependGlobalPrefix(prefix)
+	defer func() {
+		if !b.hasGlobalPrefix() {
+			return
+		}
+		for i, blob := range blobList {
+			s, _ := b.cutGlobalPrefix(blob.Name)
+			blobList[i].Name = s
+		}
+	}()
```

Comment:
If I read this correctly, by putting all of this in a defer, you are changing the cached values and calling it over and over on existing cached values.

Don't do magic in `defer`, it will bite you.